src/embezzler.ts

```diff
@@ -49,46 +51,131 @@ import { Macro, shouldRedigitize, withMacro } from "./combat";
    * @prop {(options: EmbezzlerFightOptions) => void} run This runs the combat, optionally using the provided location and macro. Location is used only by draggable fights.
-   * @prop {Requirement[]} requirements This is an array of requirements to do this fight
-   * @prop {boolean} [draggable=false] This tells the script if it is able the embezzler to a different zone
+   * @prop {EmbezzlerFightConfigOptions} options configuration options for this fight
    *
    */
   constructor(
     name: string,
     available: () => boolean,
     potential: () => number,
-    run: (options: EmbezzlerFightOptions) => void,
-    requirements: Requirement[] = [],
-    draggable = false
+    execute: (options: EmbezzlerFightRunOptions) => void,
+    options: EmbezzlerFightConfigOptions = {}
   ) {
     this.name = name;
     this.available = available;
     this.potential = potential;
-    this.run = run;
-    this.requirements = requirements;
-    this.draggable = draggable;
+    this.execute = execute;
+    this.requirements = options.requirements ?? [];
+    this.draggable = options.draggable;
+    this.canInitializeWandererCounters = options.canInitializeWandererCounters ?? true;
+    this.gregariousReplace = options.gregariousReplace ?? false;
+    this.wrongEncounterName = options.wrongEncounterName ?? this.gregariousReplace;
   }
+
+  run(options: { macro?: Macro; location?: Location } = {}): void {
+    const fightMacro = options.macro ?? embezzlerMacro();
+    if (this.draggable) {
+      this.execute(new EmbezzlerFightRunOptions(fightMacro, this.location(options.location)));
+    } else {
+      this.execute(new EmbezzlerFightRunOptions(fightMacro));
+    }
+  }
+
+  location(location?: Location): Location {
+    const suggestion =
+      this.draggable && !location && checkUnderwater() ? $location`The Briny Deeps` : location;
+
+    if (
+      (this.draggable && !suggestion) ||
+      (this.draggable === "backup" && suggestion && suggestion.combatPercent < 100)
+    ) {
+      return determineDraggableZoneAndEnsureAccess(this.draggable);
+    }
+    return $location`Noob Cave`;
+  }
+}
+
+function checkUnderwater() {
+  // first check to see if underwater even makes sense
+  if (
+    !(get("_envyfishEggUsed") || have($item`envyfish egg`)) &&
+    (booleanModifier("Adventure Underwater") || waterBreathingEquipment.some(have)) &&
+    (booleanModifier("Underwater Familiar") || familiarWaterBreathingEquipment.some(have)) &&
+    (have($effect`Fishy`) || (have($item`fishy pipe`) && !get("_fishyPipeUsed")))
+  ) {
+    // then check if the underwater copy makes sense
+    if (mallPrice($item`pulled green taffy`) < 10000 && retrieveItem($item`pulled green taffy`)) {
+      // unlock the sea
+      if (get("questS01OldGuy") === "unstarted") {
+        visitUrl("place.php?whichplace=sea_oldman&action=oldman_oldman");
+      }
+      if (!have($effect`Fishy`) && !get("_fishyPipeUsed")) use($item`fishy pipe`);
```

Comment:
overall the checkUnderwater function seems great, much better than the implicit way we were doing this previously. my one quibble is that i think we should probably call out in some sort of user-facing FAQ that if they do not have a fishy pipe they should think about getting one. maybe the answer is a small markdown file linked in the readme that just goes over some "unexpected" (ie non-directly-meat-related) resources that garbo can use for profit if the user has it, to let people know ways they can make garbo more efficient? fishy pipes are really pretty cheap, but not everyone ever dips into the sea.